src/librustc/middle/const_eval.rs

```diff
@@ -357,11 +359,15 @@ pub enum ErrKind {
     NegateOnBinary,
     NegateOnStruct,
     NegateOnTuple,
+    NegateOnArray,
+    NegateOnRepeat,
     NotOnFloat,
     NotOnString,
     NotOnBinary,
     NotOnStruct,
     NotOnTuple,
+    NotOnArray,
+    NotOnRepeat,
```

Comment:
This feels like it should be a nested enum or something (is it done like this just for flattening?).